### DISCUSSION TITLE
How to exit neovim on Close if Neogit was the only buffer?

### DISCUSSION THREAD

**Discussion by t184256** [NeogitOrg/neogit#1490] [Q&A]
Most of the time, I want to use neogit as if it was a standalone command-line TUI tool and not a vim plugin. Fire it up with `ng`, close it with a `q`.

I can start it fine with an alias to `nvim '+lua require("neogit").open({kind="replace"})'`, but the experience of quitting it is suboptimal: I press `q`, stare at a blank buffer in surprise and reach out for a `:q<CR>` to actually quit.

I suspected that the buffer that's annoying me is an empty buffer that you get when you execute `nvim` standaline, so I attempted closing it with
`nvim '+lua require("neogit").open({kind="replace"})' '+bdelete 1'`. It no longer shows up in `:buffers`, but when I `q`, here it is again, annoying me. Same with `+bwipeout 1`.

How can I execute neogit in a way that `q` will make it quit neovim entirely if there was no other buffer open?

I've seen that [in a similar case](https://github.com/NeogitOrg/neogit/issues/275) the advice was to hook into events in order to implement autoclosing, but I don't see a closing event to hook into.

**Comment by CKolkey** [comment]
There's not an event for that, but you could probably use a regular autocmd on something like `QuitPre` to check. Just stick it in the `NeogitStatus` ftplugin file.

**Reply by t184256** [reply]
Sorry, still doesn't click for me.

* QuitPre doesn't seem to be relevant
* When I hook to anything else, like, BufEnter, and iterate over buffers, I don't know how do discern me pressing q from me, say, entering a file with \<cr>

**Comment by t184256** [comment]
`nvim '+lua require("neogit").open({kind="replace"})' '+lua vim.api.nvim_buf_set_keymap(0, "n", "q", "<cmd>q<cr>", {})'` seems to work, though I don't know how ugly is that

**Reply by CKolkey** [reply]
Seems fine if it works :) I overload `q` to close other things if they are open _before_ closing the status buffer, but thats pretty minor.
